zk/legacy_executor_verifier/legacy_executor_verifier.go

```diff
@@ -305,6 +375,24 @@ func (v *LegacyExecutorVerifier) getNextOnlineAvailableExecutor() ILegacyExecuto
 	return exec
 }
 
+func (v *LegacyExecutorVerifier) availableBlocksToProcess(innerCtx context.Context, batchNumber uint64) ([]uint64, error) {
+	tx, err := v.db.BeginRo(innerCtx)
+	defer tx.Rollback()
+	if err != nil {
+		return []uint64{}, err
+	}
```

Comment:
You need to check the error here before calling the defer as tx would be nil in the case of a bad call to `BeginRo`